Add `--pkg-config-path` to `LocationOptions`

MaxDesiatov opened the pull request:
Resolves #5815.

### Motivation:

Having a specific `--pkg-config-path` option for this means we have something that's less fragile and easier to report errors on than `PKG_CONFIG_PATH` environment variable.

### Modifications:

Added `pkgConfigDirectory` to `LocationOptions`, which is propagated to `BuildPlan` as a function or initializer argument.

### Result:

`--pkg-config-path` is available as a command-line option.

tomerd commented on Sources/XCBuildSupport/PIFBuilder.swift:
```diff
@@ -24,23 +24,15 @@ import SPMBuildCore
-        self.shouldCreateDylibForDynamicProducts = shouldCreateDylibForDynamicProducts
-        self.toolchainLibDir = toolchainLibDir
-    }
+    let pkgConfigDirectory: AbsolutePath?
```
nice cleanup

MaxDesiatov commented:
> I believe `PKG_CONFIG_PATH` is a list of paths, so if we want to provide an alternative to it, we should probably also take a list here.

Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to `[AbsolutePath]` would work with some built-in support from ArgumentParser?

> If I read this correctly, it is also not being passed to XCBuild support, yet, we should do that.

Is there anything to adjust other than this place in `PIFBuilder.swift`? https://github.com/apple/swift-package-manager/pull/5949/files#diff-4558ab2f4d59c923d54773a355350f9362890d5b982aaf79ae5854f8e93eb09fR703

neonichu commented:
> Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to [AbsolutePath] would work with some built-in support from ArgumentParser?

Not sure, but we should do whatever is the most canonical way to do this. Maybe @natecook1000 has hints

MaxDesiatov commented:
@swift-ci please smoke test

MaxDesiatov commented:
@swift-ci please smoke test

MaxDesiatov commented:
> > Would it make sense to make them comma-separated? I wonder if just changing the type on the option declaration to [AbsolutePath] would work with some built-in support from ArgumentParser?
> 
> Not sure, but we should do whatever is the most canonical way to do this.

@neonichu turns out, specifying it as array allows passing the option multiple times. I've also added tests and updated `CHANGELOG.md`, this is now ready for review.

MaxDesiatov commented:
@swift-ci smoke test

MaxDesiatov commented:
@swift-ci smoke test

buttaface commented:
I was hoping we'd get this in before the 5.8 branch tomorrow, still possible?

MaxDesiatov commented:
I'm not sure, it depends on whether anyone is available for a review.

MaxDesiatov commented:
@swift-ci smoke test

tomerd commented on CHANGELOG.md:
```diff
@@ -3,6 +3,13 @@ Note: This is in reverse chronological order, so newer entries are added to the
 Swift 5.8
 -----------
 
+* [#5949]
```
I think we should sort these

MaxDesiatov replied:
It's already in reverse chronological order, or would you like to see some other sorting?

tomerd replied:
I think the rest of the document has is non-reverse chronological order?

buttaface commented:
I can confirm that [this flag works when cross-compiling SPM for Android with the latest 5.8 Dec. 20 snapshot](https://github.com/buttaface/swift-android-sdk/actions/runs/3756387249/jobs/6382415284#step:6:20), thanks.
